EtherealCereal/EtherealCereal.swift

```diff
@@ -118,14 +118,23 @@ public extension Data {
 }
 
 public extension String {
+
     public var hexadecimalData: Data? {
-        let str: String
-        let offset = self.index(self.startIndex, offsetBy: 2)
-        if self.substring(to: offset) == "0x" {
-            str = self.substring(from: offset)
-        } else {
-            str = self
+
+        let regex: NSRegularExpression
+        do {
+            regex = try NSRegularExpression(pattern: "^(?:0x)?([a-fA-F0-9]*)$", options: .caseInsensitive)
+        } catch {
+            fatalError("Invalid regular expression pattern")
         }
+        let matches = regex.matches(in: self, range: NSMakeRange(0, self.count))
+        if (matches.count != 1) {
```

Comment:
in terms of Swift's idioms, this is probably better written as: 
```swift
guard matches.count == 1 else { return nil }
```

Or you could even combine the two guard statements (forgive me if this formatting is incorrect, I haven't had a chance to read the code standards yet) 

```swift
guard 
    matches.count == 1, 
    let firstMatch = matches.first else {
        return nil 
}
```